db/data/autotest_files/autotest_01.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+
+cat <<EOF
+<test>
+    <name>test_1</name>
+    <input>input string</input>
+    <expected>expected result value</expected>
+    <actual>actual result value</actual>
+    <marks_earned>5</marks_earned>
+    <marks_total>5</marks_total>
+    <status>pass</status>
+</test>
+<test>
+    <name>test_2</name>
+    <input>input string</input>
+    <expected>expected result value</expected>
+    <actual>actual result value</actual>
+    <marks_earned>1</marks_earned>
+    <marks_total>5</marks_total>
+    <status>fail</status>
```

Comment:
Seems like we should be enforcing an invariant that status `fail` implies marks earned = 0. @adisandro is there a reason not to do this?

But even if we permit this to be a convention rather than requirement, I'd rather not our sample test illustrate this behaviour.

Same comment for the `error` output below.

Reply:
We are enforcing it in the testers, i.e. the MarkusTester base class they use.
It's true that it is not enforced on the db side, so a test script created by instructors that doesn't use it (or this one in fact) is not constrained at all.
I think we should enforce it, but also care about the messages to display in those cases.